### DISCUSSION TITLE
How can I edit an existing action?

### DISCUSSION THREAD

**Discussion by sean-dunwoody** [community/community#26525] [Actions]
<p>I have an existing action, and I want to modify the workflow file.</p>
<p>The problem is, whenever I try to do so, I’m presented with a 404 error page?</p>
<p>My process is:</p>
<ol>
<li>Go to repo</li>
<li>Select “Actions” in the menu at the top</li>
<li>Select the action I wish to modify</li>
<li>Select the “Workflow file” tab</li>
<li>Select the “Edit” icon at the top right of the file</li>
</ol>
<p>After this, I then get the 404 error.</p>
<p>I would delete and recreate the workflow if I could, but this also doesn’t seem possible (unless I’m missing something? I can’t see any way of doing this).</p>

**Comment by airtower-luna** [comment]
<ol>
<li>If you want to use the web interface, go to the “Code” tab, not “Actions”, and click through to the right file in the .github/workflows/ directory. You should be able to use the “Edit” button there.</li>
<li>If you have a local copy of the repository, just edit the workflow file (again in .github/workflows/), commit, and push.</li>
</ol>

**Reply by theking2** [reply]
The is one of those things in the GH ui that is quite counter intuitive. In the `Actions` tab there should be at least a link to the code folder

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/sean-dunwoody">@sean-dunwoody</a> , </p>
<p>I guess you have deleted/renamed that workflow yml file or deleted the whole branch. </p>
<p>Please check the branch and workflow yml file path in workflow run “Workflow file” tab. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/fdd5d670cfcafd8c9ac492614c8fddaf209df31e.png" data-download-href="/uploads/short-url/AdwWAWygrPvb9XuFWtbY1CNDUke.png?dl=1" title="Action page.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181098163-500c688f-4709-4980-9d5e-819bd746f587.png" alt="Action page.png" data-base62-sha1="AdwWAWygrPvb9XuFWtbY1CNDUke" width="690" height="373" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/fdd5d670cfcafd8c9ac492614c8fddaf209df31e_2_690x373.png, https://aws1.discourse-cdn.com/github/optimized/2X/f/fdd5d670cfcafd8c9ac492614c8fddaf209df31e_2_1035x559.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/f/fdd5d670cfcafd8c9ac492614c8fddaf209df31e.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/fdd5d670cfcafd8c9ac492614c8fddaf209df31e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Action page.png</span><span class="informations">1184×641 24.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>And then go to Code tab of the GitHub Repo, choose the matched branch and check whether the workflow yml file is still here. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3990847b1a32f4c8d3dd1b5cfd6acb4e95257106.png" data-download-href="/uploads/short-url/8deQh1xFOTOwJyxW1sPEt2Ap3TM.png?dl=1" title="code page.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181098164-30559054-d5b0-4f68-8753-8e3bc059b7d2.png" alt="code page.png" data-base62-sha1="8deQh1xFOTOwJyxW1sPEt2Ap3TM" width="520" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3990847b1a32f4c8d3dd1b5cfd6acb4e95257106_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">code page.png</span><span class="informations">653×627 22.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>If not, you will get 404 page not found when you try editing the workflow file.  </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4482428fff1ce95ea9f20d056772066bb8d7b7db.png" data-download-href="/uploads/short-url/9M3y1ugx9LFN7xpHwemV2JS8hBx.png?dl=1" title="branch file.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181098166-7094609a-b4e9-46b7-9434-dfb58f8db4dc.png" alt="branch file.png" data-base62-sha1="9M3y1ugx9LFN7xpHwemV2JS8hBx" width="690" height="358" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/4482428fff1ce95ea9f20d056772066bb8d7b7db_2_690x358.png, https://aws1.discourse-cdn.com/github/original/2X/4/4482428fff1ce95ea9f20d056772066bb8d7b7db.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/4/4482428fff1ce95ea9f20d056772066bb8d7b7db.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4482428fff1ce95ea9f20d056772066bb8d7b7db_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">branch file.png</span><span class="informations">911×473 228 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>In my scenario, the related branch has been deleted. To be able to edit the file again, I recreated the branch and the workflow yml file. </p>
